R/palaeorotate.R

```diff
@@ -1,246 +1,478 @@
+#' - Scotese, C., & Wright, N. M. (2018). PALEOMAP Paleodigital Elevation Models
+#' (PaleoDEMs) for the
+#' Phanerozoic. PALEOMAP Project.
 #' \url{https://www.earthbyte.org/paleodem-resource-scotese-and-wright-2018/}.
 #'
-#' Wright, N., Zahirovic, S., Müller, R. D., & Seton, M. (2013). Towards community-driven paleogeographic reconstructions:
-#' integrating open-access paleogeographic and paleobiology data with plate tectonics. Biogeosciences, 10(3), 1529–1541.
-#' \doi{https://doi.org/10.5194/bg-10-1529-2013}.
-#' \cr
-#' @section Developer:
+#' - Matthews, K.J., Maloney, K.T., Zahirovic, S., Williams, S.E., Seton, M.,
+#' and Müller, R.D. (2016). Global plate boundary evolution and kinematics
+#' since the late Paleozoic, Global and Planetary Change, 146, 226-250.
+#' \doi{10.1016/j.gloplacha.2016.10.002}.
+#'
+#' - Wright, N., Zahirovic, S., Müller, R. D., & Seton, M. (2013). Towards
+#' community-driven paleogeographic
+#' reconstructions: integrating open-access paleogeographic and paleobiology
+#' data with plate tectonics.
+#' Biogeosciences, 10(3), 1529–1541. \doi{10.5194/bg-10-1529-2013}.
+#'
+#' - Seton, M., Müller, R.D., Zahirovic, S., Gaina, C., Torsvik, T.H.,
+#' Shephard, G., Talsma, A., Gurnis, M., Turner, M., Maus, S., Chandler, M.
+#' (2012). Global continental and ocean basin reconstructions since 200 Ma.
+#' Earth-Science Reviews, Volume 113, Issues 3-4, July 2012, Pages 212--270.
+#' \doi{10.1016/j.earscirev.2012.03.002}.
+#'
+#' See GPlates documentation for additional information:
+#' \url{https://gwsdoc.gplates.org/reconstruction}.
+#'
+#' @section Developer(s):
 #' Lewis A. Jones
-#' @section Auditor:
-#' Missing
+#' @section Reviewer(s):
+#' Kilian Eichenseer
+#' @importFrom utils download.file
+#' @importFrom pbapply pblapply
 #' @examples
-#' \dontrun{
 #' #Generic example with a few occurrences
-#' x <- data.frame(lng = c(2, 95, 12), lat = c(46, 12, -65), age = c(88, 203, 467))
-#' palaeorotate(x = x)
+#' occdf <- data.frame(lng = c(2, -103, -66),
+#'                 lat = c(46, 35, -7),
+#'                 age = c(88, 125, 200))
+#'
+#' #Calculate palaeocoordinates using rotation files
+#' palaeorotate(occdf = occdf, method = "grid")
 #'
-#' #Now with some real fossil occurrence data! (to be updated with internal data from Bethany/Emma)
+#' #Calculate palaeocoordinates using the GPlates API
+#' palaeorotate(occdf = occdf, method = "point")
+#'
+#' #Now with some real fossil occurrence data!
 #'
 #' #Grab some data from the Paleobiology Database
-#' x <- read.csv(paste0("https://paleobiodb.org/data1.2/colls/list.csv",
-#'                      "?base_name=Scleractinia&interval=Anisian,Piacenzian"))
+#' data(tetrapods)
 #'
 #' #Assign midpoint age of fossil occurrence data for rotation
-#' x$age <- (x$max_ma + x$min_ma)/2
+#' tetrapods$age <- (tetrapods$max_ma + tetrapods$min_ma)/2
 #'
 #' #Rotate the data
-#' palaeorotate(x = x)
+#' palaeorotate(occdf = tetrapods)
```

Comment:
This is intentionally implemented. The issue here is not with the function but the models/data. It is 'normal' that palaeocoordinates can not be generated for all occurrences because they don't intersect with a plate polygon. If this happens in the grid method and not in the point method, this is because the centroid of the cell the occurrence would be linked to does not fall in the plate polygon.